### DISCUSSION TITLE
Portainer Support

### DISCUSSION THREAD

**Discussion by VinylVault** [AzuraCast/AzuraCast#6217] [Q&A]
Hi Team

I see in the code a portainer JSON file, but is Portainer installation officially supported yet? if it is, can someone please give me a short guide how to successfully deploy

Many Thanks

**Comment by bytension** [comment]
Hi! Running great here on Ubuntu with Portainer:

1 - Go to  "container-> add container" menú,  do a search in "image" search box for "azuracast"
2- From the results list pull this image from docker.io:  ghcr.io/azuracast/azuracast:stable
3- Set it the network conf you want (Keep in mind you need to expose the required TCP ports)
4- Apply.

:)

**Reply by BusterNeece** [reply]
Hmm, this works but it doesn't use the `portainer.json` file that we built for the purpose. I haven't used newer builds of Portainer, so I'll have to see if they still support importing via JSON files like the one we have, but if so then that's probably the best way to go.

**Comment by Moonbase59** [comment]
Q: How do you handle the situation that Portainer wants to bind to port 8000 since some time (for the edge agent, a bad port decision, I think) and the first station is traditionally also at 8000?

In my Proxmox/NPM/VM config, I ended up moving Portainer to another VM (it was running on the Nginx Proxy VM at first which conflicted with NPM handling 8000 as a stream and forwarding it to the AzuraCast VM, because both Portainer and NPM wanted to bind to port 8000).

**Reply by bytension** [reply]
My portainer image exposed automatically ports between 8000 to 8XXXX so it works (even for external live transmission  webdj in port 8005).

Most problems on HTTPS with reverese proxy certificates due to paticular tcp ports used

**Comment by BusterNeece** [comment]
The frustrating thing with Portainer is it doesn't let you just deploy based on a template (like the one we have) out of the box. You either have to use their built-in templates, build your own from scratch, or bind-mount your template into their Docker filesystem. You can't just say "fetch a template from this URL, parameters and all, and use it" via their web UI. It's a very frustrating limitation that has kept us from being able to offer easy Portainer support.

There's another piece of software out there that serves a similar function to Portainer called Yacht. I'm looking into whether it would support templates like this, and if so, it could be something we could recommend.

**Reply by BusterNeece** [reply]
Looking into it, and Yacht also has similar issues where it just doesn't run AzuraCast very well out of the box. Unfortunately I can't really recommend either as good solutions for our users out of the box, and we also don't offer any automatic way of setting up AzuraCast on either. :(

**Comment by Zendrex** [comment]
I don't mean to be grave digging, however has there been progress on portainer support? I saw the Yacht template and noticed a removal of the portainer version, so I'm curious if the idea has been scrapped.

**Reply by BusterNeece** [reply]
Unfortunately, I tried getting things set up with both Portainer and Yacht and ran into a ton of issues along the way, particularly with the permissions that those tools use to run their respective containers, so I scrapped it and pulled those things from the codebase for now.

If anyone is able to get a reliable version up and running on either, I'd welcome a contribution to that effect. I know Portainer is challenging because they simply don't have a way to import Compose-based templates remotely the way we'd need; Yacht does have that, but struggles farther down the line with permissions.
